pallets/market/src/benchmarking.rs

```diff
@@ -0,0 +1,409 @@
+        assert_eq!(
+            balance_entry.free,
+            (EXISTENTIAL_DEPOSIT - (COLLATERAL * n)).into()
+        );
+        assert_eq!(balance_entry.locked, (COLLATERAL * n).into());
+
+        let balance_entry = BalanceTable::<T>::get(&client);
+        assert_eq!(balance_entry.free, (EXISTENTIAL_DEPOSIT - cost).into());
+        assert_eq!(balance_entry.locked, cost.into());
+    }
+
+    /// `n`: number of submitted deals
+    #[benchmark]
+    fn settle_deal_payments(n: Linear<1, MAX_DEALS>) {
+        let caller: T::AccountId = whitelisted_caller();
+        setup_account_balance::<T>(caller.clone());
+        // Register the caller as a storage provider
+        pallet_storage_provider::Pallet::<T>::register_storage_provider(
+            RawOrigin::Signed(caller.clone()).into(),
+            BoundedVec::try_from("placeholder".as_bytes().to_vec()).unwrap(),
+            RegisteredPoStProof::StackedDRGWindow2KiBV1P1,
+        )
+        .unwrap();
+
+        // Setup a client account
+        let (client, client_pair): (T::AccountId, _) = generate_benchmark_account("client");
+        setup_account_balance::<T>(client.clone());
+
+        let (proposals, cost) =
+            prepare_proposals::<T>(n, caller.clone(), client.clone(), client_pair);
+        let proposals = BoundedVec::try_from(proposals).unwrap();
+
+        let storage_provider: OriginFor<T> = RawOrigin::Signed(caller.clone()).into();
+        Pallet::<T>::publish_storage_deals(storage_provider.clone(), proposals.clone()).unwrap();
+
+        // Run to 1 to get VRF randomness
+        run_to_block::<T>(1);
+
+        proposals
+            .into_iter()
+            .enumerate()
+            // Create SectorPreCommitInfo from the proposals
+            .map(|(idx, p)| {
+                SectorPreCommitInfoBuilder::<BlockNumberFor<T>>::default()
+                    .deals(vec![idx as u64])
+                    .sector_number(
+                        (idx as u32)
+                            .try_into()
+                            .expect("n only goes up to 32 so this should be ok"),
+                    )
+                    .raw_unsealed_cid(p.proposal.piece_cid.clone())
+                    .build()
+            })
+            // Chunk into `MAX_DEALS` groups, required because of the cfg flag
+            // meaning that sometimes MAX_DEALS > MAX_SECTORS_PER_CALL
+            .chunks(MAX_SECTORS_PER_CALL as usize)
+            // Convert IntoChunks to Iterator
+            .into_iter()
+            // Convert the chunks into Vecs
+            .map(Vec::from_iter)
+            // Convert each chunk into a BoundedVec
+            .map(|sectors_for_call| {
+                BoundedVec::<
+                    SectorPreCommitInfo<BlockNumberFor<T>>,
+                    ConstU32<MAX_SECTORS_PER_CALL>
+                >::try_from(sectors_for_call).unwrap()
+            })
+            // Submit each "chunk"
+            .for_each(|pre_commit_infos| {
```

Comment:
Can you remind me why do we need to `pre_commit` and `prove` in the test to `settle_deal_payments`?
How about just setting the DealState to `Active` manually?

Reply:
Ok, you convinced me, I don't care that much.